include/tvm/topi/transform.h

```diff
@@ -1886,13 +1883,16 @@ inline Tensor matrix_set_diag(const Tensor& input, const Tensor& diagonal, int k
                                        : 0,
                 sub_diag_right_align ? get_offset(input->shape[ndim], input->shape[ndim - 1] + k)
                                      : 0);
-          }
+            return k;
+          };
+          auto get_k = [&]() { return if_then_else(k1(0) == k2(0), k1(0), multi_diagonals()); };
+          k = get_k();
```

Comment:
can you just do something like `k = if_then_else(k1(0) == k2(0), k1(0), multi_diagonals());`?

Reply:
Thanks, Andrew. I have changed it.